test/methods.js

```diff
@@ -400,6 +400,32 @@ asyncTest("remote", function() {
 	strictEqual( v.element(e), true, "still invalid, because remote validation must block until it returns; dependency-mismatch considered as valid though" );
 });
 
+asyncTest("remote, pending class added to element while call outstanding", function() {
+	expect(3);
+	var e = $("#username"),
+		v = $("#userForm").validate({
+			rules: {
+				username: {
+					remote: "users.php"
+				}
+			},
+			submitHandler: function() {
+				ok( false, "submitHandler may never be called when validating only elements");
+			}
+		});
+
+	$(document).ajaxStop(function() {
```

Comment:
Instead of tis global `ajaxStop` hack, we should better pass a success callback into the remote method.

You can pass any option to the remote method which jquery supports for `$.ajax`, see example at http://jqueryvalidation.org/remote-method/